src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -0,0 +1,96 @@
+(ns puppetlabs.puppetserver.shell-utils
+  (:require [schema.core :as schema]
+            [clojure.java.io :as io])
+  (:import (com.puppetlabs.puppetserver ProcessWrapper ExecutionStubImpl)
+           (org.apache.commons.exec CommandLine DefaultExecutor PumpStreamHandler)
+           (java.io IOException ByteArrayInputStream ByteArrayOutputStream)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ExecutionResult
+  "A map that contains the details of the result of executing a command."
+  {:exit-code schema/Int
+   :stderr schema/Str
+   :stdout schema/Str})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Internal
+
+(schema/defn ^:private ^:always-validate
+  validate-command
```

Comment:
FWIW I believe our other `validate-foo` functions are explicitly marked with a trailing `!` to indicate we're using it for the side-effect (of throwing an exception)